src/main/java/com/club_board/club_board_server/service/book/BookService.java

```diff
@@ -148,4 +148,14 @@ public void checkUserIsOverdue(){
             reservationRepository.save(reservation);
         }
     }
+    public ReservationStatus checkUserBookStatus(Long id,Long userId){
+        boolean isReserved = reservationRepository.findReservedReservationByBookAndUser(id, userId).isPresent();
+        boolean isBorrowed = reservationRepository.findReservedReservationByBookAndUser(id, userId).isPresent();
+        ReservationStatus userReservationStatus=null;
+        if(isReserved)
+            userReservationStatus=ReservationStatus.RESERVED;
+        if(isBorrowed)
+            userReservationStatus=ReservationStatus.BORROWING;
+        return userReservationStatus;
+    }
```

Comment:
```suggestion
    public ReservationStatus checkUserBookStatus(Long id, Long userId) {
        boolean isReserved = reservationRepository.findReservedReservationByBookAndUser(id, userId).isPresent();
        if (isReserved)
            return ReservationStatus.RESERVED;
        
        boolean isBorrowed = reservationRepository.findBorrowedReservationByBookAndUser(id, userId).isPresent();
        if (isBorrowed)
            return ReservationStatus.BORROWING;
        
        return null;
    }
```
사소한 차이라 취향 차이일 수 있겠지만, early return 등의 방법으로 조회를 줄이거나, 한 번의 조회로 찾은 후 검사하는 것이 성능에 미미하게라도 도움이 될 수 있지 않을까 합니다. 만약 isReserved가 참이었다면 `findBorrowedReservationByBookAndUser`는 불필요한 조회니까요.

Reply:
오 좋은 방법인거 같아요! 바로 적용하겠습니다